src/test/java/com/google/api/generator/engine/ast/ClassDefinitionTest.java

```diff
@@ -107,6 +108,32 @@ public void validClassDefinition_statementsAndMethods() {
     // No exception thrown, we're good.
   }
 
+  @Test
+  public void invalidClassDefinition_implementsNullType() {
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          ClassDefinition.builder()
+              .setPackageString("com.google.example.library.v1.stub")
+              .setName("LibraryServiceStub")
+              .setScope(ScopeNode.PUBLIC)
+              .setImplementsTypes(
+                  Arrays.asList(
+                      TypeNode.withReference(ConcreteReference.withClazz(NullType.class))))
```

Comment:
Did you mean `TypeNode.NULL`?

Reply:
I try to cover two cases here. But I guess we do not need cover this one right now. I will remove it.